deploy/helm/kubecf/assets/operations/stacks.yaml

```diff
@@ -0,0 +1,178 @@
+{{- $_ := include "_config.lookup" (list $ "releases") }}
+{{- range $release_name, $release := $.kubecf.retval }}
+  {{- if $release.buildpack }}
+    {{- /* buildpack release conditions are either true or false, not complex expressions */}}
+    {{- if $release.condition }}
+    {{- /* *** Create/update "api" jobs for each buildpack (they already exist for default stack buildpacks) *** */}}
+- type: replace
+  path: /instance_groups/name=api/jobs/name={{ $release_name }}?/name
+  value: {{ $release_name | quote }}
+- type: replace
+  path: /instance_groups/name=api/jobs/name={{ $release_name }}/release?
+  value: {{ $release_name | quote }}
+- type: replace
+  path: /instance_groups/name=api/jobs/name={{ $release_name }}/properties?/quarks/bpm/processes
+  value: []
+    {{- else }}
+- type: remove
+  path: /instance_groups/name=api/jobs/name={{ $release_name }}?
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+{{- if not $.Values.features.eirini.enabled }}
+  {{- range $stack_name, $stack := $.Values.stacks }}
+    {{- if $stack.enabled }}
+      {{- /* *** Create "diego-cell" job to setup rootfs for each stack *** */}}
+- type: replace
+  path: /instance_groups/name=diego-cell/jobs/name={{ $stack_name }}-rootfs-setup?
+  value:
+    name: {{ $stack_name }}-rootfs-setup
+    release: {{ $stack_name }}
+    properties:
+      {{ $stack_name }}-rootfs:
+        trusted_certs:
+        - ((diego_instance_identity_ca.ca))
+      {{- if $.Values.features.credhub.enabled }}
+        - ((credhub_tls.ca))
+      {{- end }}
+        - ((uaa_ssl.ca))
+- type: replace
+  path: /instance_groups/name=diego-cell/jobs/name={{ $stack_name }}-rootfs-setup/properties/quarks?/bpm/processes
+  value: []
+    {{- else }}
```

Comment:
With the changes from the previous 2 comments it now looks like this:

```
{{- if not $.Values.features.eirini.enabled }}
  {{- range $stack_name, $stack := $.Values.stacks }}
    {{- if $stack.enabled }}
      {{- /* *** Create "diego-cell" job to setup rootfs for each stack *** */}}
- type: replace
  path: /instance_groups/name=diego-cell/jobs/name={{ $stack_name }}-rootfs-setup?
  value:
    name: {{ $stack_name }}-rootfs-setup
    release: {{ $stack_name }}
    properties:
      {{ $stack_name }}-rootfs:
        trusted_certs:
        - ((diego_instance_identity_ca.ca))
      {{- if $.Values.features.credhub.enabled }}
        - ((credhub_tls.ca))
      {{- end }}
        - ((uaa_ssl.ca))
      quarks:
        bpm:
          processes: []
    {{- else }}
- type: remove
  path: /instance_groups/name=diego-cell/jobs/name={{ $stack_name }}-rootfs-setup?
    {{- end }}
  {{- end }}
{{- end }}
```

I think this is now sufficiently clearer to leave it as-is. The nested template conditions are harder to read, but overall I prefer to have them indented to show the proper nesting level, than to outdent them all to the left margin.

Reply:
I agree that indenting the templating is useful here.

I was trying to do something about dedenting the non-templated part, but the result just ended up being even more confusing (to template-dedent it), so I guess the way you have it will have to do.